README.md

```diff
@@ -1,21 +1,97 @@
+# Compile instruction
+
+## Before compiling
+
+please check the [Releases](https://github.com/Franklalalala/usenauty/releases). There are pre-compiled linux and windows versions, respectively.
+
+## Compile a linux version
+
+The [CXX standard](https://en.wikipedia.org/wiki/C%2B%2B17) is set to be **17**, which means the gcc version need to be 8 or higher. Additionally, the cmake version needs to be **3.1** or higher.
+
+With these conditions, one can follow these steps:
+
+1. Load gcc and cmake modules. For example:
+
+```
+module load compiler/gcc/8.4.0
+module load compiler/cmake/3.15.6
+```
+
+2. Make sure the compiler works.
+
+```
+which gcc
+# /public/software/compiler/gnu/gcc-8.4.0/bin/gcc # The gcc path
+export  CC=/public/software/compiler/gnu/gcc-8.4.0/bin/gcc
+export  CXX=/public/software/compiler/gnu/gcc-8.4.0/bin/g++
+```
+
+3. Compile with commands as following:
+
+```
+git clone https://github.com/Franklalalala/usenauty
+cd usenauty
+mkdir build
+cd build
+cmake .. -G "Unix Makefiles"
+make
+```
+
+## Compile a Window version
+
+### The IDE way
+
+The easiest way to compile is to setup a [Visual Studio IDE](https://en.wikipedia.org/wiki/Microsoft_Visual_Studio#2017), according to [CXX 17 standard](https://en.wikipedia.org/wiki/C%2B%2B17), the version of VS needs to be 17 or higher. 
+
+After that, drag the whole project into the IDE. VS will automatically cmake this project. 
+
+Finally, open the `.path/to/project/src/cagesearch.cpp` file in IDE, and click build. 
+
+The target binary file will present in `.path/to/project/bin/` folder.
+
+### Otherwise
+
+One needs to prepare a compiler and cmake that meets requirements. See above.
+
+The compile lines are the same as above.
+
 # File Comments
+
 ## File differences with source library codes
-Changed some files from source (https://pallini.di.uniroma1.it/index.html) Version 2.7R1 (MAY 19 2019):
+
+Changed some files from [source](https://pallini.di.uniroma1.it/index.html) Version 2.7R1 (MAY 19 2019):
+
 - naututil.h:
-    For msvc use, change `<sys/time.h>` to `"times.h"` and `times.c` with Macro defined for arch choice.
+  For msvc use, change `<sys/time.h>` to `"times.h"` and `times.c` with Macro defined for arch choice.
 - gtools.h:
-    For msvc use, change `<sys/wait.h>` to `"wait.h"`
+  For msvc use, change `<sys/wait.h>` to `"wait.h"`
 - showg.c:
-    Add a compiler choice `ASLIB`
+  Add a compiler choice `ASLIB`
+
+## Simple command line test
+
+* Mode test and speed test for fullerene systems in `cmd_test/fullerene`
+
+* One dimension PBC system in `cmd_test/cabon_nano_tube`
+* Two dimension PBC system in `cmd_test/graphene_nano_ribbon`
```

Comment:
These seems "user guidance" but not file comments, or they at least should put in `# Code structure and function enhancement information` part.